File: 2-6/2-6-TS/Airplane.ts
/*
 * Class representing an airplane with basic functionality.
 * @author  Jakub Malhotra
 * @version 1.0
 * @since   2024-10-29
 */

export default class Airplane {

    /**
     * The speed of the airplane.
     * @type {number}
     */
    private speed: number;

    /**
     * Creates a new Plane instance.
     */
    constructor() {
        this.speed = 0;
    }

    /**
     * Gets the speed of the airplane.
     * @returns {number} The current speed of the airplane.
     */
    public get getSpeed(): number {
        return this.speed;
    }

    /**
     * Sets the speed of the airplane.
     * @param {number} speed - The new speed to set for the airplane.
     */
    public set setSpeed(speed: number) {
        this.speed = speed;
    }
}

File: 2-6/2-6-Java/Airplane.java
/*
 * Class representing an airplane with basic functionality.
 *
 * @author  Jakub Malhotra
 * @version 1.0
 * @since   2024-10-29
 */

public class Airplane {

    /**
     * The speed of the airplane.
     */
    private int speed;

    /**
     * Gets the speed of the airplane.
     *
     * @return The current speed of the airplane.
     */
    public int getSpeed() {
        return speed;
    }

    /**
     * Sets the speed of the airplane.
     *
     * @param newSpeed - The new speed to set for the airplane.
     */
    public void setSpeed(final int newSpeed) {
        this.speed = newSpeed;
    }
}
